### PR TITLE
Add attachments to FeatureService using 4.x

### PR DESCRIPTION
**Pull request by nhaney90**
Version 4.x of the JavaScript API does not currently offer any tools to work with attachments. However using JavaScript and the REST API we can recreate the look and functionality of the 3.x Attachment Editor.

### CODE REVIEW

File: web-js/4.x/add-attachments-with-42/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#Add Attachments To FeatureService with JavaScript 4.2
+
+##About
+FeatureServices can be configured to allow files to be attached to individual features in the service. In the 3.x JavaScript API methods to retrieve, add and delete attachments were provided as part of the FeatureLayer class. Additionally widgets such as the attachment and attribute inspector allowed attachments to be viewed and edited. In the 4.x API no widgets or methods have been provided to work with attachments at this time. However it is still possible to directly query and edit attachments using the ArcGIS REST API. This sample shows how to work with attachments within a 4.x application.
+
+![This is where an GIF should be. Sorry you can't see it. Try using Chrome](AddAttachments.gif "Application Demo")
+
+[Live Sample](https://nhaney90.github.io/add-attachments-with-42/index.html)
```
Comment (jgravois) [inline web-js/4.x/add-attachments-with-42/README.md]: https://esri.github.io/add-attachments-with-42/index.html

File: web-js/4.x/add-attachments-with-42/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#Add Attachments To FeatureService with JavaScript 4.2
+
+##About
+FeatureServices can be configured to allow files to be attached to individual features in the service. In the 3.x JavaScript API methods to retrieve, add and delete attachments were provided as part of the FeatureLayer class. Additionally widgets such as the attachment and attribute inspector allowed attachments to be viewed and edited. In the 4.x API no widgets or methods have been provided to work with attachments at this time. However it is still possible to directly query and edit attachments using the ArcGIS REST API. This sample shows how to work with attachments within a 4.x application.
```
Comment (jgravois) [inline web-js/4.x/add-attachments-with-42/README.md]: language like *4.x* and *'at this time'* will become stale.  if would be better to just consistently refer to *4.2*.

File: web-js/4.x/add-attachments-with-42/README.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+```javascript
+var html = "<div class='attachment-header'>Attachments:";
+for(var i = 0; i < json.attachmentInfos.length; i++) {
+	html += "<div class='attachment-item'><div class='attachment-name'>" + json.attachmentInfos[i].name + "</div><div class='download-attachment' value='" + url + "/attachments/" + json.attachmentInfos[i].id + "'></div><div class='delete-attachment' value='" + url + "/deleteAttachments?f=json&attachmentIds=" + json.attachmentInfos[i].id + "'></div></div>";
+}
+html += "</div>";
+view.popup.content.innerHTML += html;
+```
+
+To download an attachment I use a bit of trick. First I create a hyperlink element. The reference of this hyperlink is the url of the attachment on the ArcGIS Server. The hyperlink is then appended to the body of the document. The hyperlink is unseen by the user. The "click" event of the hyperlink is then triggered causing the file to be downloaded. The link is then removed from the body and deleted.
```
Comment (jgravois) [inline web-js/4.x/add-attachments-with-42/README.md]: > To download an attachment I use a bit of trick.

its a bit weird that you switch to first person tense here.

**Comment by nohe427** [comment]
I think I fixed it

File: web-js/create-svg-patches-from-esri-symbols/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#Create SVG patches from Esri symbols
```
Comment (jgravois) [inline web-js/create-svg-patches-from-esri-symbols/README.md]: what is this?

Comment (nhaney90) [inline web-js/create-svg-patches-from-esri-symbols/README.md]: A mistake. It should not have been included with the pull request that I made.

**Comment by jgravois** [comment]
the extraneous commits were the result of @nhaney90 syncing his own `master` branch with Esri/`gh-pages` upstream.  

now that @nohe427 has updated the base branch in this PR to be Esri/`gh-pages`, the commits that are not present in Esri/`master` no longer appear in the diff.

File: web-js/create-svg-patches-from-esri-symbols/README.md (hunk 4 above)
Comment (jgravois) [inline web-js/create-svg-patches-from-esri-symbols/README.md]: no worries.  i assumed it was an old artifact of doug e. fresh.

Comment (nhaney90) [inline web-js/create-svg-patches-from-esri-symbols/README.md]: I removed it

**Comment by jgravois** [comment]
@nhaney90 how do you feel about using [calcite](https://developers.arcgis.com/javascript/latest/guide/esri-icon-font/index.html) icons?

File: web-js/create-svg-patches-from-esri-symbols/README.md (hunk 4 above)
Comment (jgravois) [inline web-js/create-svg-patches-from-esri-symbols/README.md]: 👍

**Comment by nhaney90** [comment]
@jgravois changed images in popup to Calcite icons. Also updated the gif to reflect the changes.

**Comment by jgravois** [comment]
squashed and merged.  nice work @nhaney90!

**Pull request merged**